src/main/java/soot/util/BitVector.java

```diff
@@ -310,6 +312,9 @@ public String toString() {
    * @param set
    *          a bit set.
    */
+  public static long[] tmp_b;
```

Comment:
Why is is this static? You can have more than instance of BitVector.

Reply:
Yes, the program can has more than instance of BitVector, but after profiling with our tool, we found that sometimes the computation process generates some same variable (e.g., b, e, ret). So if the BitVector contents are the same with the BitVector from last iteration, we can avoid some computations here.

Reply:
I am pretty sure this would break multi threaded applications in a spectacular way. And given that analyses are often run multi threaded I'm afraid this is not a good idea.

Reply:
I agree with @MarcMil. Your proposed change would make it impossible to have two threads that each use their own BitVector, as these BitVectors may suddenly interfere with each other. Further, I haven't yet seen any convincing data that this redundance is actually a problem in practice. I would be quite surprised if this were a major cause of Soot's memory consumption. Even if so, we would need a more complex solution that can deal with multi-threading.

In total, I think that not a lot is to gain here, but quite a lot to break. I'll therefore close this PR.